src/lang/proc_macros/controller.rs

```diff
@@ -126,8 +131,9 @@ impl ProcMacroClientController {
         // Otherwise we can get messages from old client after initialization of new one.
         self.channels.clear_all();
 
-        if let Some(plugin_suite) = self.plugin_suite.take() {
-            db.remove_plugin_suite(plugin_suite);
+        for (crate_id, plugins) in self.crate_plugin_suites.clone() {
```

Comment:
This is not equal to previous semantic, `.take()` is also clearing var in place.
```suggestion
        for (crate_id, plugins) in std::mem::take(&mut self.crate_plugin_suites) {
```